packages/react/src/components/Provider/Provider.tsx

```diff
@@ -37,192 +37,176 @@ export interface ProviderProps extends ChildrenComponentProps {
 
-    const rtlProps: { dir?: 'rtl' | 'ltr' } = {}
-    // only add dir attribute for top level provider or when direction changes from parent to child
-    if (
-      !this.context ||
-      (this.context.rtl !== this.outgoingContext.rtl && _.isBoolean(this.outgoingContext.rtl))
-    ) {
-      rtlProps.dir = this.outgoingContext.rtl ? 'rtl' : 'ltr'
+    return () => {
+      if (props.target) {
+        tryCleanupWhatInput(props.target)
+      }
     }
+  }, [])
+
+  return (
+    <RendererProvider
+      renderer={outgoingContext.renderer}
+      {...{ rehydrate: false, targetDocument: outgoingContext.target }}
+    >
+      <ThemeProvider theme={outgoingContext} overwrite>
+        <ProviderBox as={as} variables={variables} {...unhandledProps} {...rtlProps}>
+          {children}
+        </ProviderBox>
+      </ThemeProvider>
+    </RendererProvider>
+  )
+}
 
-    return (
-      <RendererProvider
-        renderer={this.outgoingContext.renderer}
-        {...{ rehydrate: false, targetDocument: this.outgoingContext.target }}
-      >
-        <ThemeProvider theme={this.outgoingContext} overwrite>
-          <ProviderBox as={as} variables={variables} {...unhandledProps} {...rtlProps}>
-            {children}
-          </ProviderBox>
-        </ThemeProvider>
-      </RendererProvider>
-    )
-  }
+Provider.displayName = 'Provider'
 
-  renderStaticStylesOnce = (mergedTheme: ThemePrepared) => {
-    const { staticStyles } = this.props.theme
-    if (!this.staticStylesRendered && staticStyles) {
-      this.renderStaticStyles(this.outgoingContext.renderer, mergedTheme)
-      this.staticStylesRendered = true
-    }
-  }
+Provider.defaultProps = {
+  theme: {},
 }
+Provider.propTypes = {
+  as: PropTypes.elementType,
+  variables: PropTypes.oneOfType([PropTypes.object, PropTypes.func]),
+  theme: PropTypes.shape({
+    siteVariables: PropTypes.object,
+    componentVariables: PropTypes.object,
+    componentStyles: PropTypes.object,
+    fontFaces: PropTypes.arrayOf(
+      PropTypes.shape({
+        name: PropTypes.string,
+        paths: PropTypes.arrayOf(PropTypes.string),
+        style: PropTypes.shape({
+          fontStretch: PropTypes.string,
+          fontStyle: PropTypes.string,
+          fontVariant: PropTypes.string,
+          fontWeight: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
+          localAlias: PropTypes.string,
+          unicodeRange: PropTypes.string,
+        }),
+      }),
+    ),
+    staticStyles: PropTypes.arrayOf(
+      PropTypes.oneOfType([PropTypes.string, PropTypes.object, PropTypes.func]),
+    ),
+    animations: PropTypes.object,
+  }),
+  renderer: PropTypes.object as PropTypes.Validator<Renderer>,
+  rtl: PropTypes.bool,
+  disableAnimations: PropTypes.bool,
+  children: PropTypes.node.isRequired,
+  overwrite: PropTypes.bool,
+  target: PropTypes.object as PropTypes.Validator<Document>,
+  telemetryRef: PropTypes.object as PropTypes.Validator<React.MutableRefObject<Telemetry>>,
```

Comment:
Previously was:
```
telemetryRef: customPropTypes.ref,
```

And it was wrong as `customPropTypes.ref` also includes a function